operator/operator.go

```diff
@@ -28,6 +34,17 @@ func (o *Operator) AddController(c Controller) {
 	o.controllers = append(o.controllers, c)
 }
 
+// PrometheusCollectors returns the prometheus metric collectors for all controllers which implement metrics.Provider
+func (o *Operator) PrometheusCollectors() []prometheus.Collector {
+	collectors := make([]prometheus.Collector, 0)
+	for _, c := range o.controllers {
+		if provider, ok := c.(metrics.Provider); ok {
+			collectors = append(collectors, provider.PrometheusCollectors()...)
```

Comment:
Hm, are we going to have metric name clashes here?

Reply:
It's possible if a user has multiple of the same controller type. I think best practice in that case would be to set the namespace in the config so that they have different metric prefixes to better identify the source controller.

Reply:
Sounds good, thanks for clarifying!